examples/performance/00_overview.ipynb

```diff
@@ -0,0 +1,753 @@
+   "metadata": {},
+   "source": [
+    "## The `advanced` mode\n",
+    "\n",
+    "As already explained, `advanced` is the default optimization level in Devito. This mode performs several compilation passes to optimize the `Operator` for computation (number of flops), working set size, and data locality.\n",
+    "\n",
+    "In the next paragraphs we dissect the `advanced` mode and analyze, one by one, some of its key passes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Loop blocking\n",
+    "\n",
+    "The next cell creates a new `Operator` that adds loop blocking to what we had in `op0_omp`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "op1_omp = Operator(eq, opt=('blocking', {'openmp': True}))\n",
+    "print(op1_omp)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**\\*\\* Remark \\*\\***\n",
+    "\n",
+    "`'blocking'` is **not** an optimization level -- it rather identifies a very specific compilation pass. In other words, the `advanced` mode represents a set of passes, and `blocking` is one such pass.\n",
+    "\n",
+    "**\\*\\*\\*\\***"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `blocking` pass creates additional loops over blocks. In this simple `Operator` there is just one loop nest, so only a pair of additional loops are created. In more complex `Operator`s, several loop nests may individually be blocked, whereas others may be left unblocked -- this is decided by the Devito compiler according to certain heuristics. The size of a block is represented by the symbols `x0_blk0_size` and `y0_blk0_size`, which are runtime parameters akin to `nthreads`. \n",
+    "\n",
+    "By default, Devito applies 2D blocking and sets the default block shape to 8x8. There are two ways to set a different block shape:\n",
+    "\n",
+    "* passing an explicit value. In the example below we would run with a 24x8 block shape\n",
+    "\n",
+    "```\n",
+    "op1_omp.apply(..., x0_blk0_size=24)\n",
+    "```\n",
+    "\n",
+    "* letting the autotuner run for a while, looking for a better block shape. There are several autotuning modes. A short summary is available [here](https://github.com/devitocodes/devito/wiki/FAQ#devito_autotuning)\n",
+    "\n",
+    "```\n",
+    "op1_omp.apply(..., autotune='aggressive')\n",
+    "```\n",
+    "\n",
+    "Loop blocking also provides two optimization options:\n",
+    "\n",
+    "* `blockinner={False, True}` -- to enable 3D (or any nD, n>2) blocking\n",
+    "* `blocklevels={int}` -- to enable hierarchical blocking, to exploit the cache hierarchy \n",
+    "\n",
+    "In the example below, we construct an `Operator` featuring six-dimensional loop blocking, where the first three loops represent outer blocks, whereas the second three loops represent inner blocks within an outer block."
```

Comment:
`featuring six-dimensional loop blocking`
Is this a good expression to use?

Reply:
improving